doc/rfcs/003-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,147 @@
+
+- Teletype could allow you to enter a person's GitHub username (or their email address) to invite them to your portal. This would remove the need for sharing a URL via a third-party service in order to collaborate, and it would reduce the need to provide a list of past collaborators (i.e., you could just enter a username each time you want to collaborate). However, it would make it possible for any GitHub user to cause invitations to appear inside your Atom instance. This introduces a vector for harassment, so we're avoiding this approach.
+- Teletype could allow you to invite other users that you're already associated with in some way (e.g., fellow collaborators on a GitHub repository, fellow members of a GitHub organization or team). This would remove the need for sharing a URL via a third-party service in order to collaborate, and it would reduce the need to provide a list of past collaborators, and it would reduce the harassment vector described in the previous bullet. However, it would introduce tradeoffs that we'd prefer to avoid:
+    - Teletype would need additional permissions (i.e., [OAuth scopes](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/#available-scopes)) to fetch the list of users you're associated with. (Today, Teletype  uses a scopeless token and has no access to your private information.)
+    - Some users belong to organizations with thousands of members. Just because you're a member of the same organization as someone else doesn't mean that you're comfortable seeing portal invitations from them.
+- Teletype could add support for audio before adding support for inviting past collaborators to your portal. When collaborating with people in different physical locations, you'll likely need audio support in order to collaborate. Since Teletype doesn't currently provide audio support, these collaborators will still need a third-party solution for audio. If you're already relying on a third-party solution for audio, you can often use that same solution to send your collaborators your portal ID, which reduces the need for Teletype providing a list of past collaborators. However, there are still many [scenarios where it's helpful to quickly invite a past collaborator even if Teletype doesn't yet support audio](https://github.com/atom/teletype/pull/344#discussion_r175319913).
+
+##### What is the impact of not doing this?
+
+People will collaborate less often. Given the additional steps needed to start collaborating, there will be more instances where people decide that it's not worth the effort (i.e., people will enjoy rich collaboration less often).
+
+## Unresolved questions
+
+##### What unresolved questions do you expect to resolve through the RFC process before this gets merged?
+
+- If I invite a past collaborator to join my portal, and that person has multiple Atom windows open, can we show the invitation only in the frontmost window?
```

Comment:
We could potentially show the pending invitation in all of their windows, and honestly we're going to have a hard time doing anything else without coming up with some kind of RPC mechanism between package instances. Slightly better would be to show the invitation in all windows, then dismiss it from all windows once it is accepted in one of them. That could probably happen via the server somehow.

Reply:
Yeah, I guess a clean way of modeling this would be to introduce the notion of an invitation on the server that can be canceled somehow, and send a pub-sub notification to the invitees saying that the invitation has expired. The side benefit of doing this is also that hosts could be allowed to cancel pending invites, in case they've clicked on the wrong person.

Reply:
I am concerned about showing this to all windows, as a security precaution. If I'm signed in to multiple computers, I would only want to receive the invitation on the active window